src/test/java/seedu/address/model/person/BirthdateTest.java

```diff
@@ -0,0 +1,75 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class BirthdateTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Birthdate(null));
+    }
+
+    @Test
+    public void constructor_invalidDate_throwsIllegalArgumentException() {
+        String invalidBirthdate = "2001.09.11";
+        assertThrows(IllegalArgumentException.class, () -> new Birthdate(invalidBirthdate));
+    }
+
+    @Test
+    void isValidBirthdate() {
+        // null birthdate
+        assertThrows(NullPointerException.class, () -> Birthdate.isValidBirthdate(null));
+
+        // invalid birthdate
+        assertFalse(Birthdate.isValidBirthdate("")); // empty string
+        assertFalse(Birthdate.isValidBirthdate(" ")); // spaces only
+        assertFalse(Birthdate.isValidBirthdate("02022002")); // incorrect date format
+        assertFalse(Birthdate.isValidBirthdate("11.11.2001")); // incorrect date format
+        assertFalse(Birthdate.isValidBirthdate("2009.09.21")); // incorrect date format
+        assertFalse(Birthdate.isValidBirthdate("10 Jul 2022")); // incorrect date format
+        assertFalse(Birthdate.isValidBirthdate("12 12 2007")); // incorrect date format
+        assertFalse(Birthdate.isValidBirthdate("2004-11-11")); // incorrect date format
+
+        // valid birthdate
+        assertTrue(Birthdate.isValidBirthdate("01-01-2001"));
+        assertTrue(Birthdate.isValidBirthdate("12-02-1927"));
+        assertTrue(Birthdate.isValidBirthdate("31-12-2003"));
+    }
```

Comment:
Are we able to combine these tests, given that `IllegalArgumentException` is thrown only when `Birthdate#isValidBirthdate` returns false after being called in the `#checkArgument` within the Birthdate constructor?

Reply:
Following the convention set by the other test classes of a Person's field, such as `PhoneTest`, `AddressTest` and `NameTest`, the constructor's test case is usually separated from the `isValid__()` test case, despite `isValid__()` method being called in `checkArgument()` in each constructor. 

To be consistent with the other test classes, I think it might be better to separate the test cases for the constructor and the `isValidBirthdate()` method.

Open for inputs.